lib/super_digit.rb

```diff
@@ -1,15 +1,19 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - log10(n)
+# Space Complexity - log10(n)
 def super_digit(n)
-   
+  # base case
+  return n if n/10 == 0
+  
+  # recursive case
+  return super_digit(n/10 + n%10)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - k*log10(n)
+# Space Complexity - k*log10(n)
 def refined_super_digit(n, k)
-    
-end
-    
\ No newline at end of file
+  concatenated = "#{n}" * k
+  return super_digit(concatenated.to_i)
```

Comment:
Consider a dynamic programming solution

```suggestion
  sum = super_digit(n)
  sum *= k
  return super_digit(sum)
```